### DISCUSSION TITLE
Error: (error "line 66 column 36: Sort mismatch at argument #1 for function (declare-fun or (Bool Bool) Bool) supplied sort is Int") sat

### DISCUSSION THREAD

**Discussion by ghost** [Z3Prover/z3#5680] [Q&A]
I have just started workign with Z3 for my bachelors computer science and engineering. One of the assignments was to provide solutions for the 8 queens problem. 

The logic behind the solutions can be found below:

![image](https://user-images.githubusercontent.com/78276074/142739773-90d7717a-a13b-4f3a-aa9a-0ca1dcb4d03a.png)

The text  file attaches is the logic written out for Z3 but I can't seem to figure out what causes the error. I was hoping someone could give me some insight since I am still a novice.

[8queens.txt](https://github.com/Z3Prover/z3/files/7575376/8queens.txt)

**Comment by NikolajBjorner** [comment]
The error message suggests that you are passing an integer expression to "or". 
Your text file declares the variables as Int. You should try to declare them as Bool.

**Comment by NikolajBjorner** [comment]
See also https://github.com/Z3Prover/z3/tree/master/examples/userPropagator
for a tuned version of 8 queens enumeration
